### PR TITLE
feat(a11y): add cdkAriaLive directive

### PR DESCRIPTION
**Pull request by mmalerba**
unblocks #11144

### CODE REVIEW

File: src/cdk/a11y/live-announcer/live-announcer.spec.ts [hunk 1]
```diff
@@ -111,6 +113,76 @@ describe('LiveAnnouncer', () => {
...
+  const invokeMutationCallbacks = () => mutationCallbacks.forEach(cb => cb());
+
+  beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [A11yModule],
+      declarations: [DivWithCdkAriaLive],
+      providers: [{
+        provide: MutationObserverFactory,
+        useValue: {
+          create: function(callback: Function) {
```
Comment (jelbourn) [inline src/cdk/a11y/live-announcer/live-announcer.spec.ts]: Arrow function?

File: src/cdk/a11y/live-announcer/live-announcer.ts [hunk 2]
```diff
@@ -81,12 +87,52 @@ export class LiveAnnouncer implements OnDestroy {
...
 
+/**
+ * A directive that works similarly to aria-live, but uses the LiveAnnouncer to ensure compatibility
+ * with a wider range of browsers and screen readers.
+ */
+@Directive({
+  selector: '[cdkAriaLive]'
+})
+export class CdkAriaLive implements OnDestroy {
+  @Input('cdkAriaLive')
```
Comment (jelbourn) [inline src/cdk/a11y/live-announcer/live-announcer.ts]: Add user-facing JsDoc

File: src/cdk/a11y/live-announcer/live-announcer.ts [hunk 3]
```diff
@@ -81,12 +87,52 @@ export class LiveAnnouncer implements OnDestroy {
...
+  get politeness(): AriaLivePoliteness { return this._politeness; }
+  set politeness(value: AriaLivePoliteness) {
+    this._politeness = value === 'polite' || value === 'assertive' ? value : 'off';
+    if (this._politeness === 'off') {
+      if (this._subscription) {
+        this._subscription.unsubscribe();
+        this._subscription = null;
+      }
+    } else {
+      this._subscription = this._ngZone.runOutsideAngular(
```
Comment (jelbourn) [inline src/cdk/a11y/live-announcer/live-announcer.ts]: Shouldn't you only set up a new subscription if there isn't already one?

**Comment by jelbourn** [comment]
Looks like circle still has a failure

**Comment by mmalerba** [comment]
@DevVersion thinks this directive may not be necessary. I'll experiment with getting the datepicker a11y working without it and then decide what to do with this PR

**Comment by mmalerba** [comment]
@DevVersion I found that just setting `aria-atomic` and `aria-live` was not enough to make it work in VoiceOver for my calendar use case. So will go ahead with this PR.

**Comment by DevVersion** [comment]
@mmalerba Sounds good.

**Pull request merged**